src/api/app/models/status_message.rb

```diff
@@ -1,12 +1,51 @@
 class StatusMessage < ApplicationRecord
-  belongs_to :user
+  belongs_to :user # TODO: rename as creator
+  has_many :status_message_acknowledgements, dependent: :destroy
+  has_many :users, through: :status_message_acknowledgements
+
   validates :user, :severity, :message, presence: true
+
   scope :alive, -> { where(deleted_at: nil).order('created_at DESC') }
+  scope :announcements, -> { alive.where(severity: 'announcement') }
+
+  enum severity: { information: 0, green: 1, yellow: 2, red: 3, announcement: 4 }
+  enum communication_scope: { all_users: 0, logged_in_users: 1, admin_users: 2, in_beta_users: 3, in_rollout_users: 4 }
+
+  # xml: A Nokogiri object
+  def self.from_xml(xml)
+    StatusMessage.create! if xml.blank?
+    doc = Nokogiri::XML(xml, &:strict).root
+    message = doc.css('message').text
+    severity = doc.css('severity').text
+    scope = doc.css('scope').text
+    scope = 'all_users' if scope.blank?
+    StatusMessage.new(message: message, severity: severity, communication_scope: scope, user: User.session!)
+  end
 
   def delete
     self.deleted_at = Time.now
     save
   end
+
+  def acknowledge!
+    users << User.session!
+  end
+
+  def self.newest_announcement_for_current_user
+    announcement = StatusMessage.announcements.find_by(communication_scope: StatusMessage.communication_scopes_for_current_user)
```

Comment:
I would prefer to raise an exception here than return nil. Nil checking is a bad smell.

Reply:
I will refactor it later.